### DISCUSSION TITLE
Separate vendor projects from solution

### DISCUSSION THREAD

**Discussion by superewald** [cpm-cmake/CPM.cmake#360] [Q&A] [answered]
Hey there, first of all thanks for this awesome tool! 
I have a simple question which probably is not related to CPM itself but I just can't get my head around it.

When running the `CMakelists.txt` below on windows (using VS generator) the resulting *.sln contains alot of dependency projects of gRPC.
I know those "third-party" projects are required in order to build my project but the solution really blows with that ~50 projects. 

What's the recommended way to separate those third-party projects from my solution ? 
In the end, I want a solution or vcxproj which contains only the targets I included and linked to, not their dependencies.
Pretty sure I missed something crucial in the docs or somewhere, sorry if so.

<details><summary><b>CMakeLists.txt</b></summary>

```
cmake_minimum_required(VERSION 3.14 FATAL_ERROR)

project(
    CPM_SUBP_TEST
    LANGUAGES CXX
)

include(cmake/CPM.cmake)

CPMAddPackage(
  Name grpc
  GITHUB_REPOSITORY grpc/grpc
  VERSION 1.46.3
  EXCLUDE_FROM_ALL YES
  OPTIONS
    "gRPC_BUILD_TESTS OFF"
)

add_executable(CPM_SUBP_TEST main.cpp)
target_link_libraries(CPM_SUBP_TEST PUBLIC grpc)
```

</details>

<details><summary><b>VS Solution</b></summary>

![cpm_subp_test_grpc_sln](https://user-images.githubusercontent.com/34782055/173024077-d00f771f-d21c-4bd5-b7a4-664d82ca8e1e.PNG)

</details>

**Comment by OlivierLDff** [comment, marked as answer]
You can 
```cmake
if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
  set_property(GLOBAL PROPERTY USE_FOLDERS ON)
endif()
```

And then do something like:
```cmake
set_target_properties(absl_bad_optional_access PROPERTIES FOLDER "Dependencies")
```

for each dependencies.

And maybe have a look if you can contribute for them to add such option in related project, I don't know if absl have for example a `ABSL_TARGET_FOLDER` or something like that.

**Reply by superewald** [reply]
Awesome, thanks for the quick answer 👍
